recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,1418 @@
+        # See notes above, VTK may do even deeper checks but this hopefully will be good enough.
+        #
+        if self.options.build_all_modules:
+            for option in options_to_check:
+                if not _is_module_NO(option):
+                    return True # something was not 'NO', then this module is at least partially enabled
+            # ALL are NO, then assume this module is disabled
+            return False
+        else:
+            for option in options_to_check:
+                if _is_module_YES_or_WANT(option):
+                    return True
+            return False
+
+
+    # special case for QT as it is more involved
+    @property
+    def _is_any_Qt_enabled(self):
+        return self._is_module_enabled([
+            self.options.group_enable_Qt,
+            self.options.module_enable_GUISupportQt,
+            self.options.module_enable_GUISupportQtQuick,
+            self.options.module_enable_GUISupportQtSQL,
+            self.options.module_enable_RenderingQt,
+            self.options.module_enable_ViewsQt
+            ])
+
+
+    # Required as VTK adds 'd' to the end of library files on windows
+    @property
+    def _lib_suffix(self):
+        return "d" if self.settings.os == "Windows" and self.settings.build_type == "Debug" else ""
+
+    def build_requirements(self):
+        self.tool_requires("sqlite3/3.41.1")
+
+    def source(self):
+        if self.options.use_source_from_git:
+            self.run("git clone -b release --single-branch " + self.git_url + " " + self.source_folder)
+            # note: we give the branch a name so we don't have detached heads
+            # TODO change to standard git and python chdir
+            # Version: support targeting a commit hash instead of a version tag, assume version number < 8 chars long
+            # 1.34.67 ... only 7 chars long
+            # anything more, assume a branch name or a hash (or short hash)
+            git_hash = self.version if len(self.version) > 7 else "v" + self.version
+            self.run("cd " + self.source_folder + " && git checkout -b branch-" + git_hash + " " + git_hash)
+        else:
+            get(self, **self.conan_data["sources"][self.version],
+                    strip_root=True,
+                    destination=self.source_folder)
+
+    def _third_party(self):
+        parties = {
+                # LEFT field:  target name for linking, will be used as TARGET::TARGET in package_info()
+                # RIGHT field: package/version to require
+                "cli11":             "cli11/2.3.2",
+                "double-conversion": "double-conversion/3.2.1",
+                "eigen":             "eigen/3.4.0",
+                "expat":             "expat/2.5.0",
+                "exprtk":            "exprtk/0.0.1",   # TODO upgrade to 0.0.2 (there was a problem with first attempt)
+                "fmt":               "fmt/8.1.1",      # VTK 9.1.0 release docs mention a PR - confirmed merged 8.1.0
```

Comment:
fmt can be updated to 9.1.0 by backporting more commits

https://gitlab.kitware.com/vtk/vtk/-/commit/121585b9ff051f640ef8f76fca450c69bbf56ef7.patch
https://gitlab.kitware.com/vtk/vtk/-/commit/8493b4e946d5853aa72cc7096e7ebab286a6f20a.patch

Reply:
Ok, but I'd probably just push to upgrade VTK to a newer version that has those commits?    Also I have some changes for this recipe in the pipeline so it works with conan v2.  Currently conan v2 complains about requirements that aren't used in the final components in the package.

Reply:
Note to self: future VTK has fmt bumped to 10.1.1 to deal with MSVC 2022 C++17